snowshu/adapters/source_adapters/snowflake_adapter.py

```diff
@@ -367,16 +367,47 @@ def _validate_key_index_error(self,
                 f"Remote key {remote_key} not found in {relation.temp_dot_notation} table."
             ) from err
 
+    def format_remote_key(
+        self, relation: Relation, local_key: str, remote_key: str
+    ) -> str:
+        """Formats the remote key based on whether it needs to be quoted or not."""
+
+        local_key_type_query = (
```

Comment:
It seems like we can put 375-387 lines inside the `if` statement. So the SQL which find `local_key_type` will be executed only in the case when we need it, instead of always.